ee/server/service/teams.go

```diff
@@ -1244,6 +1251,25 @@ func (svc *Service) editTeamFromSpec(
 	return nil
 }
 
+func validateTeamCustomSettings(invalid *fleet.InvalidArgumentError, prefix string, customSettings []fleet.MDMProfileSpec) {
+	for i, prof := range customSettings {
+		count := 0
+		for _, b := range []bool{len(prof.Labels) > 0, len(prof.LabelsIncludeAll) > 0, len(prof.LabelsExcludeAny) > 0} {
+			if b {
+				count++
+			}
+		}
+		if count > 1 {
+			invalid.Append(fmt.Sprintf("%s_settings.custom_settings", prefix),
+				fmt.Sprintf(`Couldn't edit %s_settings.custom_settings. For each profile, only one of "labels_exclude_any", "labels_include_all" or "labels" can be included.`, prefix))
+		}
+		if len(prof.Labels) > 0 {
+			customSettings[i].LabelsIncludeAll = customSettings[i].Labels
+			customSettings[i].Labels = nil
```

Comment:
Note that it's important not to set `prof.LabelsIncludeAll` or `prof.Labels` but to assign directly to the index in the slice, because the slice contains structs, not pointers to structs (assigning `prof.<field>` would not be visible to the caller).